library/alloc/src/raw_vec.rs

```diff
@@ -559,12 +576,10 @@ unsafe impl<#[may_dangle] T, A: Allocator> Drop for RawVec<T, A> {
 
 // Central function for reserve error handling.
 #[cfg(not(no_global_oom_handling))]
-#[inline]
-fn handle_reserve<T>(result: Result<T, TryReserveError>) -> T {
-    match result.map_err(|e| e.kind()) {
-        Ok(res) => res,
-        Err(CapacityOverflow) => capacity_overflow(),
-        Err(AllocError { layout, .. }) => handle_alloc_error(layout),
+fn handle_reserve_err(e: TryReserveError) -> ! {
```

Comment:
Tiny nit, just so that the function name rhymes with `TryReserveError` and `handle_alloc_error`
```suggestion
fn handle_reserve_error(e: TryReserveError) -> ! {
```